packages/grid/x-data-grid-premium/src/hooks/features/aggregation/gridAggregationUtils.ts

```diff
@@ -0,0 +1,413 @@
+  if (groupingParams.treeDepth > 1) {
+    groupingParams.ids.forEach((parentId) => {
+      const parentNode = tree[parentId];
+      if (parentNode.depth === groupingParams.treeDepth - 1) {
+        return;
+      }
+
+      addGroupFooter(parentNode);
+    });
+  }
+
+  return {
+    ...groupingParams,
+    ids,
+    idRowsLookup,
+    tree,
+  };
+};
+
+/**
+ * Compares two sets of aggregation rules to determine if they are equal or not.
+ */
+export const hasAggregationRulesChanged = (
+  previousValue: GridAggregationRules | undefined,
+  newValue: GridAggregationRules,
+) => {
+  const previousFields = Object.keys(previousValue ?? {});
+  const newFields = Object.keys(newValue);
+
+  if (!isDeepEqual(previousFields, newFields)) {
+    return true;
+  }
+
+  const hasAggregationRuleChanged = (
+    previousRule: GridAggregationRule | undefined,
+    newRule: GridAggregationRule | undefined,
+  ) => {
+    if (previousRule?.aggregationFunction !== newRule?.aggregationFunction) {
+      return true;
+    }
+
+    if (previousRule?.aggregationFunctionName !== newRule?.aggregationFunctionName) {
+      return true;
+    }
+
+    return false;
+  };
+
+  return newFields.some((field) => {
+    const previousColumnAggregationRules = previousValue?.[field];
+    const newColumnAggregationRules = newValue[field];
+
+    return (
+      hasAggregationRuleChanged(
+        previousColumnAggregationRules?.inline,
+        newColumnAggregationRules.inline,
+      ) ||
+      hasAggregationRuleChanged(
+        previousColumnAggregationRules?.footer,
+        newColumnAggregationRules.footer,
+      )
+    );
+  });
+};
+
+export const getGroupingColumns = ({
+  apiRef,
+  columnsLookup,
+}: {
+  apiRef: React.MutableRefObject<GridApiPremium>;
+  columnsLookup: GridColumnRawLookup;
+}): {
+  groupingCriteria?: string[];
+  groupingColumnField: string;
+}[] => {
+  if (columnsLookup[GRID_TREE_DATA_GROUPING_FIELD]) {
+    return [{ groupingColumnField: GRID_TREE_DATA_GROUPING_FIELD }];
+  }
+
+  if (columnsLookup[GRID_ROW_GROUPING_SINGLE_GROUPING_FIELD]) {
+    return [
+      {
+        groupingColumnField: getRowGroupingFieldFromGroupingCriteria(null),
+      },
+    ];
+  }
+
+  return gridRowGroupingModelSelector(apiRef)
+    .map((groupingCriterion) => ({
+      groupingCriteria: [groupingCriterion],
+      groupingColumnField: getRowGroupingFieldFromGroupingCriteria(groupingCriterion),
+    }))
+    .filter(({ groupingColumnField }) => !!columnsLookup[groupingColumnField]);
+};
+
+/**
+ * Returns the label to render for the current footer
+ * The priority is as follows:
+ * 1. props.aggregationFooterLabel (if defined)
+ * 2. l10n `aggregationMultiFunctionLabel` (if several aggregation function used)
+ * 3. aggregationFunction.label (if defined)
+ * 4. l10n `aggregationFunctionLabel${capitalize(aggregationFunctionName)}` (if defined)
+ * 5. ''
+ */
+export const getAggregationFooterLabel = ({
```

Comment:
@m4theushw 

I added support for aggregation footer label as asked

It does not support aggregation without grouping yet (finding the column on which to apply the aggregation is a pain and maybe we can just add a `aggregationLabelField` prop so that the users says on which columns he wants to put the label.

But for aggregation with grouping (row grouping or tree data), I can put the label on the grouping columns.
Below you have the function to determine the label to show.
I think there are to many options but I don't know which one to remove.
Maybe remove 1. and make 2. a function that takes the params of 1.

Side note: the try / catch is to avoid crashing when the locale does not have the label for the aggregation function. On the filter label it does crash and I think we should have a cleaner fallback (maybe throw an error, but an explicit one, not the generic non-existing locale error)

Reply:
(the doc and tests are not done yet, I want to decide the basic behaviors before)